kong/db/init.lua

```diff
@@ -547,12 +547,24 @@ do
         if run_up then
           -- kong migrations bootstrap
           -- kong migrations up
-          ok, err = self.connector:run_up_migration(mig.name,
-                                                    strategy_migration.up)
-          if not ok then
-            self.connector:close()
-            return nil, fmt_err(self, "failed to run migration '%s' up: %s",
-                                mig.name, err)
+          if strategy_migration.up and strategy_migration.up ~= "" then
+            ok, err = self.connector:run_up_migration(mig.name,
+                                                      strategy_migration.up)
+            if not ok then
+              self.connector:close()
+              return nil, fmt_err(self, "failed to run migration '%s' up: %s",
+                                  mig.name, err)
+            end
+          end
+
+          if strategy_migration.up_f then
+            ok, err = self.connector:run_up_migration(mig.name,
```

Comment:
I see in https://github.com/Kong/kong/pull/6608/files#diff-13a8c7acbd78b8151e140a4d391bd6f5639f126ca5d6c2501b214af616b69e1bR603-R604

we simply called

```lua
xpcall(f, debug.traceback, self.connector, mig_helpers)
```

for executing the `teardown` callbacks. Can we do the same for `strategy_migration.up_f`? this seems to avoid the special handling needed inside the connector code.

Reply:
Good point! I extracted the `pcall` to `kong/db/init.lua`. The connector has no changes now, looks much better.